### PR TITLE
fix: better read_group input validation checking: group and hints fields

### PR DESCRIPTION
**Pull request by alamb**
This is a PR towards ensuring the proper implementation of the read_group gRPC call (https://github.com/influxdata/influxdb_iox/issues/448)

It adds validation to the input parameters of `read_group` per the semantics described in [the read_group mini spec](https://docs.google.com/document/d/1UoCov4_NEhJHTxQftn5-YlzfhCiP5hgqDq9ikbRRa3g/)

Here is the rough implementation plan:

- [x] Implement read_group for `None` aggregates (https://github.com/influxdata/influxdb_iox/pull/538)
- [x] Error if any hints are provided, or if there is a discrepancy between `GroupBy` and `GroupNone` type (this PR)
- [ ] Implement read_group for `Sum`, `Count` and `Mean` aggregates
- [ ] Implement user defined aggregates for `min`, `max`, `first` and `last`, selector functions
- [ ] Implement read_group for `Min`, `Max`, `First` and `Last` (using selctor functions)

### CODE REVIEW

File: src/server/rpc/service.rs [hunk 1]
```diff
@@ -1898,13 +1909,43 @@ mod tests {
...
         // ---
-        // test error
+        // test error hit in request processing
         // ---
         let request = ReadGroupRequest {
             read_source: source.clone(),
             range: None,
             predicate: None,
-            group_keys: vec![],
+            group_keys: vec![String::from("tag1")],
```
Comment (e-dard) [inline src/server/rpc/service.rs:1918]: There are a couple of different approaches to coercing a `str` into `String` in this PR (`.into()` and `String::from`). I filed [an issue](https://github.com/influxdata/influxdb_iox/issues/553) to start a discussion about agreeing on a preferred method to do this in the code-base. FWIW I think `to_owned` is probably a good choice (I'm guilty of using `to_string` a lot though!!).

**Review by e-dard — APPROVED** [review]
LGTM!

File: src/server/rpc/service.rs (hunk 1 above)
Comment (alamb) [reply src/server/rpc/service.rs:1918]: I have no excuse for being inconsistent in this PR -- I will at least be consistent in this one while we sort out https://github.com/influxdata/influxdb_iox/issues/553. Thank you for pointing it out

File: src/server/rpc/expr.rs [hunk 2]
```diff
@@ -473,9 +484,19 @@ fn build_binary_expr(op: Operator, inputs: Vec<Expr>) -> Result<Expr> {
 
 pub fn make_read_group_aggregate(
     aggregate: Option<RPCAggregate>,
-    _group: RPCGroup,
+    group: RPCGroup,
     group_keys: Vec<String>,
```
[the remark this answers is not in the document]
Comment (alamb) [reply src/server/rpc/expr.rs:488]: This is an excellent point. I looked into this suggestion and it turns out the `Vec<String>` is used further down to create `GroupByAndAggregate` which does need a `Vec<String> `. When I dug deeper  into figuring out if I could avoid that, I realized that something very cool (and Rusty) is going on:

The Vec<String> is coming from the RPC request and this code is effectively passing along ownership of those Strings from the gPRC request to the query layer (without any additional `clone`). So from what I can see short of fixing the `tonic` code not to make `Vec<String>` in the gRPC request structures this pattern is already avoiding any allocations.

**Pull request merged**